polars/polars-lazy/polars-plan/src/dsl/python_udf.rs

```diff
@@ -0,0 +1,156 @@
+use std::io::Cursor;
+use std::sync::Arc;
+
+use polars_arrow::error::PolarsResult;
+use polars_core::datatypes::{DataType, Field};
+use polars_core::error::*;
+use polars_core::prelude::Series;
+use pyo3::types::{PyBytes, PyModule};
+use pyo3::{PyErr, PyObject, Python};
+
+use super::expr_dyn_fn::*;
+use crate::constants::MAP_LIST_NAME;
+use crate::prelude::*;
+
+// Will be overwritten on python polar start up.
+pub static mut CALL_LAMBDA: Option<fn(s: Series, lambda: &PyObject) -> PolarsResult<Series>> = None;
+pub(super) const MAGIC_BYTE_MARK: &[u8] = "POLARS_PYTHON_UDF".as_bytes();
+
+pub struct PythonFunction {
+    python_function: PyObject,
+    output_type: Option<DataType>,
+}
+
+impl PythonFunction {
+    pub fn new(lambda: PyObject, output_type: Option<DataType>) -> Self {
+        Self {
+            python_function: lambda,
+            output_type,
+        }
+    }
+
+    pub(crate) fn try_deserialize(buf: &[u8]) -> PolarsResult<Arc<dyn SeriesUdf>> {
+        debug_assert!(buf.starts_with(MAGIC_BYTE_MARK));
+        // skip header
+        let buf = &buf[MAGIC_BYTE_MARK.len()..];
+        let mut reader = Cursor::new(buf);
+        let output_type: Option<DataType> =
+            ciborium::de::from_reader(&mut reader).map_err(map_err)?;
+
+        let remainder = &buf[reader.position() as usize..];
+
+        Python::with_gil(|py| {
+            let func: PyObject = PyModule::from_code(
+                py,
+                r#"
+def pickle_python_udf(arg):
+    import pickle
+    return pickle.loads(arg)
+"#,
+                "",
```

Comment:
@alexander-beedie note that this creates a python module called `""`. Maybe that gets registered in the python process?

Reply:
Definitely connected; interestingly the `__init__.py` seemed to knock the errors down from 1 to 2, but that may be a coincidence 🤷   Looks like an artefact of tracing the expected error. I can dig in at home in a few hours.